db/migrations/2003110348_extract_results_from_attempts.sql

```diff
@@ -0,0 +1,423 @@
+-- +migrate Up
+RENAME TABLE `attempts` TO `results`;
+
+ALTER TABLE `results`
+    DROP FOREIGN KEY `fk_attempts_creator_id_users_group_id`,
+    DROP CHECK `cs_attempts_order`,
+    DROP CHECK `cs_attempts_score_computed_is_valid`,
+    ADD CONSTRAINT `cs_results_score_computed_is_valid` CHECK ((`score_computed` between 0 and 100)),
+    DROP CHECK `cs_attempts_score_edit_value_is_valid`,
+    ADD CONSTRAINT `cs_results_score_edit_value_is_valid` CHECK ((ifnull(`score_edit_value`,0) between -(100) and 100)),
+    RENAME COLUMN `group_id` TO `participant_id`,
+    RENAME INDEX `group_item` TO `participant_id_item_id`,
+    RENAME INDEX `group_item_score_desc_score_obtained_at` TO `participant_id_item_id_score_desc_score_obtained_at`,
+    ADD COLUMN `attempt_id` BIGINT(20) NOT NULL DEFAULT 0 AFTER `participant_id`;
+
+DROP TRIGGER `before_insert_attempts`;
+
+CREATE TABLE `attempts` (
+    `participant_id` BIGINT(20) NOT NULL,
+    CONSTRAINT `fk_attempts_participant_id_groups_id` FOREIGN KEY (`participant_id`) REFERENCES `groups`(`id`) ON DELETE CASCADE,
+    `id` BIGINT(20) NOT NULL
+        COMMENT 'Identifier of this attempt for this participant, 0 is the default attempt for the participant, the next ones are sequentially assigned.',
+    PRIMARY KEY (`participant_id`, `id`),
+    `creator_id` BIGINT(20) DEFAULT NULL COMMENT 'The user who created this attempt',
+    CONSTRAINT `fk_attempts_creator_id_users_group_id` FOREIGN KEY (`creator_id`) REFERENCES `users`(`group_id`) ON DELETE SET NULL,
+    `parent_attempt_id` BIGINT(20) DEFAULT NULL COMMENT 'The attempt from which this one was forked. NULL for the default attempt.',
```

Comment:
if you do not put this `participant_id`, `parent_attempt_id` constraint for the reason we discussed, you still have to set `parent_attempt_id` to `attempts.id`

Reply:
fixed

Reply:
After thinking about this again, I believe this foreign key doesn't make any sense :( The parent_attempt_id may exist for some different participant, making MySQL act as everything is okay. Probably it would be better to revert the change :/

Reply:
reverted